### DISCUSSION TITLE
Aris Studio Development - Clean up commit history

### DISCUSSION THREAD

**Discussion by kiraio-moe** [Tualin14/baPlayer#21] [Development]
Hi @Tualin14, been long time not in here.

You see, I have made a massive blunder while trying to rebase my commits. While trying to push my rebased changes, it's says that I didn't have force push permission, so I try to pull and merge with the remote. Without checking things before pushing to repo, I realized I  made a mistake and flooding the commit history. Sorry 😞

So, should we clean things up first or it's a bad idea to rebase in a community (public) repository?

**Comment by ttttttttttttttllllllllllllll** [comment]
Did this behavior affect the normal operation of the program or did it cause other catastrophic effects? 
If there is no impact on the program's operation, There is no need to deal with it or you deal with it in a way you think is better.

Since I need to finish my dissertation by June, I won't tackle anything other than my dissertation until mid-June at the latest. Sorry.

**Reply by kiraio-moe** [reply]
> Did this behavior affect the normal operation of the program or did it cause other catastrophic effects?

Nope, it didn't cause any effects on the program, just a messy commit history.

> If there is no impact on the program's operation, There is no need to deal with it or you deal with it in a way you think is better.

Well, I fear if I doing rebase & force-pushing myself there will be causing some difficulties on your side. I did read some articles on this matter and yeah, it's hard to do it in open repository.

On the time being, I will just cleaning up commit history on my forked repo.

> Since I need to finish my dissertation by June, I won't tackle anything other than my dissertation until mid-June at the latest. Sorry.

Good luck on that, pal :)

**Reply by ttttttttttttttllllllllllllll** [reply]
It did affect my git, so I created a dev branch to commit development content and merged it into the master branch for each release.

**Reply by kiraio-moe** [reply]
> It did affect my git, so I created a dev branch to commit development content and merged it into the master branch for each release.

Sorry for the trouble 🙏😅
